src/main/java/omero/gateway/facility/BrowseFacility.java

```diff
@@ -547,6 +548,39 @@ public IObject findIObject(SecurityContext ctx, IObject o)
         return Collections.emptyList();
     }
 
+    /**
+     * Load project by name
+     *
+     * @param ctx
+     *            The {@link SecurityContext}
+     * @param name
+     *            The name of the project
+     *
+     * @return A collection of {@link ProjectData}s
+     * @throws DSOutOfServiceException
+     *             If the connection is broken, or not logged in
+     * @throws DSAccessException
+     *             If an error occurred while trying to retrieve data from OMERO
+     *             service.
+     */
+    public Collection<ProjectData> getProjects(SecurityContext ctx, String name) throws DSOutOfServiceException, DSAccessException {
+        try {
+            IQueryPrx proxy = gateway.getQueryService(ctx);
+            StringBuilder sb = new StringBuilder();
+            ParametersI param = new ParametersI();
+            param.add("name", omero.rtypes.rstring(name));
+            sb.append("select p from Project as p ");
+            sb.append("where p.name = :name");
+            List<IObject> results = proxy.findAllByQuery(sb.toString(), param);
```

Comment:
This looks ideal for `findAllByString`.

Reply:
(Ah, but that doesn't hydrate and wrap the projects it returns, sorry: that's probably why you do it like this.)

But then why not just "select p.id" rather than extracting the IDs afterward?